librosa/util/utils.py

```diff
@@ -627,8 +636,26 @@ def normalize(S, norm=np.inf, axis=0):
     else:
         raise ParameterError('Unsupported norm: {}'.format(repr(norm)))
 
+    # get the vectors-ids smaller than the threshold
+    small_values_idx = (length < threshold)
+
     # Avoid div-by-zero
-    length[length < SMALL_FLOAT] = 1.0
+    length[small_values_idx] = 1.0
+
+    if mode and np.sum(small_values_idx):
+        # init empty slice object
+        slices = [slice(None)] * S.ndim
+
+        # set slice for target-axis
+        target_axis_idx = int(np.logical_not(axis))
```

Comment:
I don't understand this bit.  Can you explain in comments?

It seems like this will limit the function to only work on 2d arrays, whereas before it worked for arbitrary dimension (even if the comments did not say as much).  I'd like to not lose that functionality if we don't have to.